net/devif/devif_callback.c

```diff
@@ -258,11 +235,31 @@ FAR struct devif_callback_s *
                        FAR struct devif_callback_s **list_tail)
 {
   FAR struct devif_callback_s *ret;
+  int i;
 
   /* Check the head of the free list */
 
   net_lock();
-  ret  = g_cbfreelist;
+  ret = g_cbfreelist;
+
+  /* Allocate the callback entry from heap */
```

Comment:
@anchao can you please add part of https://github.com/apache/incubator-nuttx/pull/5110#issuecomment-1002741733 as an explanation here?

Reply:
Of course, let us wait for the conclusion of [#5104](https://github.com/apache/incubator-nuttx/pull/5104), I think this pull request needs to be further enhanced

Reply:
Let's apply the similar change as #5104

Reply:
Done